it/src/test/java/com/google/cloud/teleport/it/bigquery/DefaultBigQueryResourceManagerTest.java

```diff
@@ -0,0 +1,373 @@
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.internal.verification.Times;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+/** Unit tests for {@link com.google.cloud.teleport.it.bigquery.DefaultBigQueryResourceManager}. */
+@RunWith(JUnit4.class)
+public class DefaultBigQueryResourceManagerTest {
+
+  @Rule public final MockitoRule mockito = MockitoJUnit.rule();
+
+  @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+  private BigQuery bigQuery;
+
+  @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+  private Dataset dataset;
+
+  @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+  private Table table;
+
+  @Mock private Credentials credentials;
+  @Mock private Page<Table> tables;
+  @Mock private Schema schema;
+  @Mock private InsertAllRequest.RowToInsert rowToInsert;
+  @Mock private TableResult tableResult;
+  @Mock private FieldValueList rowToRead;
+  @Mock private InsertAllResponse insertResponse;
+  @Mock private Map<Long, List<BigQueryError>> writeErrorsMap;
+
+  private static final String TABLE_NAME = "table-name";
+  private static final String DATASET_ID = "dataset-id";
+  private static final String TEST_ID = "test-id";
+  private static final String PROJECT_ID = "test-project";
+  private static final String REGION = "us-central1";
+
+  private DefaultBigQueryResourceManager testManager;
+
+  @Before
+  public void setUp() {
+    testManager =
+        DefaultBigQueryResourceManager.builder(TEST_ID, PROJECT_ID).setBigQuery(bigQuery).build();
+  }
+
+  private void prepareCreateDataset() {
+    when(bigQuery.create((DatasetInfo) any())).thenReturn(dataset);
+    Mockito.lenient().when(dataset.getDatasetId().getDataset()).thenReturn(DATASET_ID);
+  }
+
+  private void prepareCreateTable() {
+    when(dataset.get(any())).thenReturn(table);
+    when(bigQuery.getTable(any())).thenReturn(null);
+    Mockito.lenient().when(table.getTableId().getTable()).thenReturn(TABLE_NAME);
+  }
+
+  @Test
+  public void testBuilderSetsValuesCorrectly() {
+    DefaultBigQueryResourceManager.Builder tmBuilder =
+        DefaultBigQueryResourceManager.builder(TEST_ID, PROJECT_ID);
+    assertThat(tmBuilder.getDatasetId())
+        .matches(TEST_ID.replace('-', '_') + "_\\d{8}_\\d{6}_\\d{6}");
+    tmBuilder
+        .setProjectId("fake_project_id")
+        .setTestId("fake_test_id")
+        .setDatasetId(DATASET_ID)
+        .setCredentials(credentials)
+        .setBigQuery(bigQuery);
+
+    assertThat(tmBuilder.getCredentials()).isEqualTo(credentials);
+    assertThat(tmBuilder.getProjectId()).isEqualTo("fake_project_id");
+    assertThat(tmBuilder.getTestId()).isEqualTo("fake_test_id");
+    assertThat(tmBuilder.getDatasetId()).matches(DATASET_ID);
+    assertThat(tmBuilder.getBigQuery()).isEqualTo(bigQuery);
+
+    DefaultBigQueryResourceManager tm = tmBuilder.build();
+
+    assertThat(tm.getProjectId()).isEqualTo("fake_project_id");
+    assertThat(tm.getTestId()).isEqualTo("fake_test_id");
```

Comment:
Do we care aboure `testId` in the resource manager? AFAIU it's really used anywhere. I'd remove the `testId` instance variable and this check in the test.